sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/RolePermissions.java

```diff
@@ -187,81 +173,75 @@ public boolean equals(Object o) {
 
         RolePermissions that = (RolePermissions) o;
 
-        if (read != that.read) {
+        if (readPermission != that.readPermission) {
             return false;
         }
-        if (write != that.write) {
+        if (writePermission != that.writePermission) {
             return false;
         }
-        return execute == that.execute;
+        return executePermission == that.executePermission;
     }
 
     @Override
     public int hashCode() {
-        int result = (read ? 1 : 0);
-        result = 31 * result + (write ? 1 : 0);
-        result = 31 * result + (execute ? 1 : 0);
+        int result = (readPermission ? 1 : 0);
+        result = 31 * result + (writePermission ? 1 : 0);
+        result = 31 * result + (executePermission ? 1 : 0);
         return result;
     }
 
     /**
-     * Grants the role read permissions on the resource.
-     *
-     * @return Whether or not the role has read permissions on the resource.
+     * @return the read permission status
      */
-    public boolean read() {
-        return read;
+    public boolean hasReadPermission() {
+        return readPermission;
     }
 
     /**
-     * Grants the role write permissions on the resource.
-     *
-     * @return Whether or not the role has write permissions on the resource.
+     * @return the write permission status
      */
-    public boolean write() {
-        return write;
+    public boolean hasWritePermission() {
+        return writePermission;
     }
 
     /**
-     * Grants the role execute permissions on the resource.
-     *
-     * @return Whether or not the role has execute permissions on the resource.
+     * @return the execute permission status
      */
-    public boolean execute() {
-        return execute;
+    public boolean hasExecutePermission() {
+        return executePermission;
     }
 
     /**
-     * Grants the role read permissions on the resource.
+     * Sets the read permission status.
      *
-     * @param read {@code true} if read permissions should be granted.
-     * @return The updated RolePermissions object.
+     * @param setReadPermission Permission status to set
+     * @return the updated RolePermissions object
      */
-    public RolePermissions read(boolean read) {
-        this.read = read;
+    public RolePermissions setReadPermission(boolean setReadPermission) {
```

Comment:
boolean setReadPermission -> hasReadPermission

Reply:
I think setters should be set and getters should be has (this is how we do it for SASPermission)

Reply:
Ohhhh nvm

Reply:
done